pkg/kubectl/cmd/set/set_limit.go

```diff
@@ -0,0 +1,234 @@
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set limit will NOT contact api-server but run locally")
+	cmdutil.AddRecordFlag(cmd)
+	cmd.Flags().BoolVar(&options.Remove, "remove", false, "If true, set limit will remove resource limit/requests from pod template")
+	cmd.Flags().StringVar(&options.Limits, "limits", options.Limits, "Here is the limit usage stuff")
+	cmd.Flags().StringVar(&options.Requests, "requests", options.Requests, "Here is the limit usage stuff")
+	cmdutil.AddRecursiveFlag(cmd, &options.Recursive)
+	return cmd
+}
+
+func (o *LimitOptions) Complete(f *cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.Mapper, o.Typer = f.Object(false)
+	o.UpdatePodSpecForObject = f.UpdatePodSpecForObject
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = cmdutil.GetFlagString(cmd, "output") == "name"
+	o.Record = cmdutil.GetRecordFlag(cmd)
+	o.ChangeCause = f.Command()
+	o.PrintObject = f.PrintObject
+	o.Cmd = cmd
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	builder := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, o.Recursive, o.Filenames...).
+		Flatten()
+	if !o.Local {
+		builder = builder.
+			SelectorParam(o.Selector).
+			ResourceTypeOrNameArgs(o.All, args...).
+			Latest()
+	}
+
+	o.Infos, err = builder.Do().Infos()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *LimitOptions) Validate() error {
+	var err error
+	//if len(o.Limits) == 0 && len(o.Requests) == 0 && o.Remove == false {
+	if (len(o.Limits) == 0 && len(o.Requests) == 0) != o.Remove {
+		return fmt.Errorf("you must specify an update to limits or requests (in the form of --limits/--requests or --remove but not both)")
+	}
+
+	o.ResourceRequirements, err = kubectl.HandleResourceRequirements(map[string]string{"limits": o.Limits, "requests": o.Requests})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *LimitOptions) Run() error {
+	allErrs := []error{}
+	patches := CalculatePatches(o.Infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		transformed := false
+		_, err := o.UpdatePodSpecForObject(info.Object, func(spec *api.PodSpec) error {
+			containers, _ := selectContainers(spec.Containers, o.ContainerSelector)
+			if len(containers) != 0 {
+				for i := range containers {
+					containers[i].Resources = o.ResourceRequirements
+					transformed = true
+				}
+			} else {
+				allErrs = append(allErrs, fmt.Errorf("error: unable to find container named %s", o.ContainerSelector))
+			}
+			return nil
+		})
+		return transformed, err
+	})
+
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			allErrs = append(allErrs, fmt.Errorf("error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err))
+			continue
+		}
+
+		//no changes
```

Comment:
@smarterclayton I believe that I put the continue where you meant. If I understand correctly an empty patch is not an error so I added a note to stderr when a patch is empty that the resource was not changed